llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -5110,57 +5288,79 @@ bool SROA::splitAlloca(AllocaInst &AI, AllocaSlices &AS) {
   NumAllocaPartitions += NumPartitions;
   MaxPartitionsPerAlloca.updateMax(NumPartitions);
 
-  // Migrate debug information from the old alloca to the new alloca(s)
-  // and the individual partitions.
```

Comment:
Useful comment surely?

Reply:
To my mind a reader going through the function top-to-bottom will want to know what this lambda is doing, hence it's good to have a comment where it's defined. This is a style thing, YMMV.